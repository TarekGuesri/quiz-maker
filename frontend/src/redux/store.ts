import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import ui from "./ui/ui-slice";

export const store = configureStore({
  reducer: {
    ui,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
