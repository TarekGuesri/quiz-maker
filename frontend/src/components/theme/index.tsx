import React, { FC } from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import PropTypes from 'prop-types';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { useAppSelector } from 'src/redux/hooks';
import varelaRound from 'src/assets/fonts/VarelaRound-Regular.ttf';

export const Theme: FC = (props) => {
  const { darkMode } = useAppSelector((state) => state.ui);

  const theme = createTheme({
    palette: {
      mode: darkMode ? 'dark' : 'light',
      background: {
        default: darkMode ? '#333' : '#f7f7f7',
        paper: darkMode ? '#333' : '#fbfbfb',
      },
      primary: {
        main: '#8e49de',
        // contrastText:
      },
      secondary: {
        main: '#f9de4a',
      },
      contrastThreshold: 3,
      tonalOffset: 0.2,
    },
    typography: {
      fontFamily: 'Varela Round',
    },
    components: {
      MuiCssBaseline: {
        styleOverrides: `
          @font-face {
            font-family: 'Varela Round';
            font-style: normal;
            font-display: swap;
            font-weight: 400;
            src: url(${varelaRound}) format('truetype');
        `,
      },
    },
  });

  return (
    <ThemeProvider theme={theme}>
      {props.children}
      <CssBaseline />
    </ThemeProvider>
  );
};

Theme.propTypes = {
  children: PropTypes.node.isRequired,
};
