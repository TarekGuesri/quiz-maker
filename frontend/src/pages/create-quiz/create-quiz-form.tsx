import React from "react";
import { useAppDispatch, useAppSelector } from "src/redux/hooks";
import { createStyles, makeStyles } from "@mui/styles";
import { Theme } from "@mui/material";
import Alert from "@mui/material/Alert";
import Divider from "@mui/material/Divider";
import Grid from "@mui/material/Grid";
import TextField from "@mui/material/TextField";
import Typography from "@mui/material/Typography";
import Radio from "@mui/material/Radio";
import RadioGroup from "@mui/material/RadioGroup";
import FormControlLabel from "@mui/material/FormControlLabel";
import FormControl from "@mui/material/FormControl";
import FormLabel from "@mui/material/FormLabel";
import Pagination from "@mui/material/Pagination";
import Button from "@mui/material/Button";
import Tooltip from "@mui/material/Tooltip";
import {
  addQuestion,
  changeAnswer,
  changeTitle,
  changePage,
  changeQuestion,
  createQuiz,
  removeQuestion,
  setSelectedAnswer,
} from "src/redux/create-quiz/create-quiz-slice";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    formLabel: {
      color: theme.palette.text.primary,
      fontSize: "1.2rem",
      "&.Mui-focused": {
        color: theme.palette.text.primary,
      },
    },
    helperText: {
      color: theme.palette.text.secondary,
      marginBottom: theme.spacing(2),
    },
    formControlLabel: {
      marginBottom: theme.spacing(2),
    },
    pagination: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
    },
    button: {
      width: "190px",
      padding: "10px",
    },
    buttonGridItem: {
      paddingLeft: "0!important",
    },
  }),
);

export const CreateQuizForm: React.FC = ({}) => {
  const classes = useStyles();
  const dispatch = useAppDispatch();
  const {
    title,
    questionIndex,
    questions,
    selectedAnswers,
    isLoading,
    errorMessage,
  } = useAppSelector((state) => state.createQuiz);

  const handleSelectAnswer = (event: React.ChangeEvent<HTMLInputElement>) => {
    const selectedAnswer = (event.target as HTMLInputElement).value;

    dispatch(setSelectedAnswer(selectedAnswer));
  };

  const handleChangePage = (
    _event: React.ChangeEvent<unknown>,
    value: number,
  ) => {
    if (value) {
      dispatch(changePage(value - 1));
    }
  };

  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value;

    dispatch(changeTitle(value));
  };

  const handleChangeQuestion = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value;

    dispatch(changeQuestion(value));
  };

  const handleChangeAnswer = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;

    dispatch(changeAnswer({ name, value }));
  };

  const currentQuestion = questions[questionIndex];

  // We use this for selected answer's radio button
  const selectedAnswer = selectedAnswers[questionIndex];

  return (
    <>
      <Typography variant="h5" component="div" mb={4}>
        Create a quiz
      </Typography>

      <Grid container>
        {/* Quiz Title */}
        <Grid item xs={12} mb={3}>
          <TextField
            value={title}
            onChange={handleChangeTitle}
            label="Quiz Title"
            variant="outlined"
            size="medium"
            fullWidth
          />
        </Grid>

        {/* Quiz Title */}
        <Grid item xs={12} mb={3}>
          <Divider />
        </Grid>

        {/* Question */}
        <Grid item xs={12} mb={3}>
          <TextField
            value={currentQuestion.content}
            onChange={handleChangeQuestion}
            label={`Question ${questionIndex + 1}`}
            variant="outlined"
            size="medium"
            fullWidth
          />
        </Grid>

        {/* Answers */}
        <Grid item xs={12} mb={2}>
          <FormControl component="fieldset">
            <FormLabel className={classes.formLabel}>
              Answers
              <Typography className={classes.helperText}>
                Check the correct answer
              </Typography>
            </FormLabel>

            <RadioGroup
              aria-label="gender"
              name="controlled-radio-buttons-group"
              value={selectedAnswer}
              onChange={handleSelectAnswer}
            >
              {currentQuestion.answers.map((answer, index) => (
                <FormControlLabel
                  key={answer.id}
                  className={classes.formControlLabel}
                  value={answer.id}
                  control={<Radio />}
                  label={
                    <TextField
                      label={`Answer ${index + 1}...`}
                      name={answer.id}
                      value={answer.text}
                      onChange={handleChangeAnswer}
                    />
                  }
                />
              ))}
            </RadioGroup>
          </FormControl>
        </Grid>

        {/* Pagination */}
        <Grid item xs={12} mb={4}>
          <Pagination
            className={classes.pagination}
            count={questions.length}
            page={questionIndex + 1}
            color="primary"
            onChange={handleChangePage}
          />
        </Grid>

        {/* Error Message */}
        {errorMessage && (
          <Grid item xs={12} mb={4}>
            <Alert variant="filled" severity="error">
              {errorMessage}
            </Alert>
          </Grid>
        )}
        {/* Buttons */}
        <Grid
          container
          item
          direction="row"
          xs={12}
          maxWidth={700}
          mx="auto"
          spacing={3}
        >
          <Grid item xs={12} md={4} className={classes.buttonGridItem}>
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              onClick={() => dispatch(removeQuestion())}
              disabled={questions.length < 2}
            >
              Remove Question
            </Button>
          </Grid>
          <Grid item xs={12} md={4} className={classes.buttonGridItem}>
            <Tooltip title="You can only have 10 questions max per quiz">
              <div>
                <Button
                  variant="contained"
                  color="primary"
                  className={classes.button}
                  onClick={() => dispatch(addQuestion())}
                  disabled={questions.length >= 10}
                >
                  Add a question
                </Button>
              </div>
            </Tooltip>
          </Grid>
          <Grid item xs={12} md={4} className={classes.buttonGridItem}>
            <Button
              variant="contained"
              color="secondary"
              className={classes.button}
              onClick={() => dispatch(createQuiz())}
              disabled={isLoading}
            >
              Create Quiz
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </>
  );
};
