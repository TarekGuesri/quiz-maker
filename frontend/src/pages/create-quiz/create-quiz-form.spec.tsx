import React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { Theme } from "src/components/theme";
import { store } from "src/redux/store";

import { CreateQuizForm } from "./create-quiz-form";

describe("src/components/create-quiz-form.tsx", () => {
  afterEach(() => cleanup());

  test("Render CreateQuizForm component", () => {
    const renderedComponent = render(
      <Provider store={store}>
        <Theme>
          <CreateQuizForm />
        </Theme>
      </Provider>,
    );

    expect(renderedComponent.container).toMatchSnapshot();
  });
});
