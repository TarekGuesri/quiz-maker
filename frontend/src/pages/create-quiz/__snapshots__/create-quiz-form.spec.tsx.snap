// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/create-quiz-form.tsx Render CreateQuizForm component 1`] = `
<div>
  <div
    class="MuiTypography-root MuiTypography-h5 css-1j54lbx-MuiTypography-root"
  >
    Create a quiz
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ouds40-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-une3nw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          Quiz Title
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1q0ijwp-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-dhh0px"
            >
              <span>
                Quiz Title
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ouds40-MuiGrid-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ouds40-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-une3nw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          Question 1
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1q0ijwp-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-dhh0px"
            >
              <span>
                Question 1
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1sf285j-MuiGrid-root"
    >
      <fieldset
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorPrimary makeStyles-formLabel-1 css-46gqlf-MuiFormLabel-root"
        >
          Answers
          <p
            class="MuiTypography-root MuiTypography-body1 makeStyles-helperText-2 css-jqqqr0-MuiTypography-root"
          >
            Check the correct answer
          </p>
        </label>
        <div
          aria-label="gender"
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd makeStyles-formControlLabel-3 css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-10h4w5k-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="controlled-radio-buttons-group"
                type="radio"
                value="012925c6-581c-4767-b784-4991a63ab26d"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-jqqqr0-MuiTypography-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-une3nw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Answer 1...
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k82hce-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    name="012925c6-581c-4767-b784-4991a63ab26d"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-dhh0px"
                    >
                      <span>
                        Answer 1...
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd makeStyles-formControlLabel-3 css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-10h4w5k-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="controlled-radio-buttons-group"
                type="radio"
                value="4251170e-b557-4f7e-a8d7-ed0dc796fd81"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-jqqqr0-MuiTypography-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-une3nw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Answer 2...
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k82hce-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    name="4251170e-b557-4f7e-a8d7-ed0dc796fd81"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-dhh0px"
                    >
                      <span>
                        Answer 2...
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd makeStyles-formControlLabel-3 css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-10h4w5k-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="controlled-radio-buttons-group"
                type="radio"
                value="bc33804e-2a77-42f7-89e1-3c2dc2ba7f60"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-jqqqr0-MuiTypography-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-une3nw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Answer 3...
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k82hce-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    name="bc33804e-2a77-42f7-89e1-3c2dc2ba7f60"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-dhh0px"
                    >
                      <span>
                        Answer 3...
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd makeStyles-formControlLabel-3 css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-10h4w5k-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                name="controlled-radio-buttons-group"
                type="radio"
                value="e310d02f-b731-4a2f-a37c-39141bf13885"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-jqqqr0-MuiTypography-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-une3nw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Answer 4...
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k82hce-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    name="e310d02f-b731-4a2f-a37c-39141bf13885"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-dhh0px"
                    >
                      <span>
                        Answer 4...
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </span>
          </label>
        </div>
      </fieldset>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-15qc0cf-MuiGrid-root"
    >
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-text makeStyles-pagination-4 css-1oj2twp-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1gx04xy-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1gx04xy-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1gx04xy-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
    
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 css-15haxxf-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 makeStyles-buttonGridItem-6 css-17dw2p9-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled makeStyles-button-5 css-kygxw5-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Remove Question
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 makeStyles-buttonGridItem-6 css-17dw2p9-MuiGrid-root"
      >
        <div
          aria-label="You can only have 10 questions max per quiz"
          class=""
          data-mui-internal-clone-element="true"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-5 css-kygxw5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Add a question
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 makeStyles-buttonGridItem-6 css-17dw2p9-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-5 css-1g8rry4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Create Quiz
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
